corehq/apps/app_manager/suite_xml.py

```diff
@@ -1277,57 +1277,9 @@ def build_detail(self, module, detail_type, detail, detail_column_infos,
-                # add form action to detail
-                form = self.app.get_form(module.case_list_form.form_id)
-
-                d.action = Action(
-                    display=Display(
-                        text=Text(locale_id=id_strings.case_list_form_locale(module)),
-                        media_image=module.case_list_form.media_image,
-                        media_audio=module.case_list_form.media_audio,
-                    ),
-                    stack=Stack()
-                )
-                frame = PushFrame()
-                frame.add_command(XPath.string(id_strings.form_command(form)))
-
-                def get_datums_meta_for_form(form):
-                    if form.form_type == 'module_form':
-                        datums_meta = self.get_case_datums_basic_module(form.get_module(), form)
-                    elif form.form_type == 'advanced_form':
-                        datums_meta, _ = self.get_datum_meta_assertions_advanced(form.get_module(), form)
-                        datums_meta.extend(SuiteGenerator.get_new_case_id_datums_meta(form))
-                    else:
-                        raise SuiteError("Unexpected form type '{}' with a case list form: {}".format(
-                            form.form_type, form.unique_id
-                        ))
-                    return datums_meta
-
-                target_form_dm = get_datums_meta_for_form(form)
-                source_form_dm = get_datums_meta_for_form(module.get_form(0))
-                for target_meta in target_form_dm:
-                    if target_meta['requires_selection']:
-                        # This is true for registration forms where the case being created is a subcase
-                        try:
-                            [source_dm] = [
-                                source_meta for source_meta in source_form_dm
-                                if source_meta['case_type'] == target_meta['case_type']
-                            ]
-                        except ValueError:
-                            raise SuiteError("Form selected as case list form requires a case "
-                                             "but no matching case could be found: {}".format(form.unique_id))
-                        else:
-                            frame.add_datum(StackDatum(
-                                id=target_meta['datum'].id,
-                                value=session_var(source_dm['datum'].id))
-                            )
-                    else:
-                        s_datum = target_meta['datum']
-                        frame.add_datum(StackDatum(id=s_datum.id, value=s_datum.function))
-
-                frame.add_datum(StackDatum(id=RETURN_TO, value=XPath.string(id_strings.menu_id(module))))
-                d.action.stack.add_frame(frame)
+            if module.case_list_form.form_id and detail_type.endswith('short')\
+                    and not module.put_in_root:
```

Comment:
Added in the `and not module.put_in_root`. The rest is just extracting it to a function.